llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -482,11 +481,10 @@ InstructionCost RISCVTTIImpl::getShuffleCost(TTI::ShuffleKind Kind,
       //   vmsne.vi  v0, v8, 0
 
       return LT.first *
-             (TLI->getLMULCost(LT.second) + // FIXME: this should be 1 for andi
```

Comment:
If the index is non-zero, then there's a vslide needed in the current sequence which does scale with LMUL.

So, this change is wrong for non-zero index.

Reply:
For SK_Broadcast, my understanding is that the index should always be the first element of the source vector. 
I was based on this
```
      if (Shuffle->isZeroEltSplat())
        return TargetTTI->getShuffleCost(TTI::SK_Broadcast, VecTy,
                                         Shuffle->getShuffleMask(), CostKind, 0,
                                         nullptr, Operands);
```
Could we clarify this?"

Reply:
It looks like you're correct here.  There are other cases which create SK_Broadcast, but from what I can tell, all are consistent with the splat index always being zero.

Reply:
To chime in here, it's defined in TargetTransformInfo.h as being a broadcast of element zero:

```c++
enum ShuffleKind {
    SK_Broadcast,        ///< Broadcast element 0 to all other elements.
```